### PR TITLE
Add time spent in WAITING_FOR_RESOURCES state to query timeline log

### PR DESCRIPTION
**Pull request by raunaqmorarka**
Adding resource wait time of a query to query timeline log as it can be useful to have this information from logs for debugging

### CODE REVIEW

**Comment by martint** [comment]
Thanks for submitting this. We should also consider adding it to the `QueryCompletedEvent` so that event listener plugins can access it.

**Comment by raunaqmorarka** [comment]
> Thanks for submitting this. We should also consider adding it to the `QueryCompletedEvent` so that event listener plugins can access it.

@martint I've added resource waiting time to QueryStatistics to make it accessible in QueryCompletedEvent

File: presto-main/src/main/java/io/prestosql/event/QueryMonitor.java [hunk 1]
```diff
@@ -150,6 +150,7 @@ public void queryImmediateFailureEvent(BasicQueryInfo queryInfo, ExecutionFailur
                         ofMillis(0),
```
Comment (martint) [inline presto-main/src/main/java/io/prestosql/event/QueryMonitor.java]: The commit message is too long. We can shorten it while conveying the essence of the change: "Add resource waiting time to query completion event"

Comment (raunaqmorarka) [reply presto-main/src/main/java/io/prestosql/event/QueryMonitor.java]: Done

**Comment by raunaqmorarka** [comment]
> Thanks! I'll merge it once Travis passes.
@martint Travis job seems to be stuck in queued state, is it possible to re-submit it ?

**Pull request merged**